src/gui/SearchWidget.cpp

```diff
@@ -76,14 +75,9 @@ bool SearchWidget::eventFilter(QObject *obj, QEvent *event)
             }
         }
         else if (keyEvent->matches(QKeySequence::MoveToNextLine)) {
-            // If Down is pressed at EOL in the search edit, move
-            // the focus to the entry view.
-            QLineEdit* searchEdit = m_ui->searchEdit;
-            if (!searchEdit->hasSelectedText() &&
-                searchEdit->cursorPosition() == searchEdit->text().length()) {
-                emit downPressed();
-                return true;
-            }
+            // If Down is pressed move the focus to the entry view.
```

Comment:
On macOS, the cursor is considered to be positioned at EOL when the whole text is selected, so simply deleting the first condition (59962dd) would do the job.